src/wallet/direct_send.rs

```diff
@@ -133,12 +142,29 @@ impl Wallet {
             }
         };
 
-        let mut output = Vec::<TxOut>::new();
-
         let txout = {
             let value = match send_amount {
-                SendAmount::Max => (total_input_value - fee).to_sat(),
-                SendAmount::Amount(a) => a.to_sat(),
+                SendAmount::Max => {
+                    // Find dust value for destination scriptpubkey.
+                    let destination_dust = dest_addr.script_pubkey().minimal_non_dust();
+                    if total_input_value > fee + destination_dust {
+                        (total_input_value - fee).to_sat()
+                    } else {
+                        // TODO: Write error statement here.
+                        return Err(WalletError::Protocol("".to_string()));
+                    }
+                }
```

Comment:
This logic doesn't seem correct. If we are sending a value to the external user, we don't need to check for dust. Their utxo should exist anyway, weather below or above dust limit. The old logic can stay here.